src/mem.rs

```diff
@@ -227,7 +227,14 @@ impl Decompress {
         self.inner.raw.next_in = input.as_ptr() as *mut _;
         self.inner.raw.avail_in = input.len() as c_uint;
         self.inner.raw.next_out = output.as_mut_ptr() as *mut _;
-        self.inner.raw.avail_out = output.len() as c_uint;
+        self.inner.raw.avail_out = {
+            let avail_out = output.len();
+            if (avail_out > 0) && (avail_out & c_uint::MAX as usize == 0) {
+                c_uint::MAX
+            } else {
+                avail_out as c_uint
+            }
+        };
```

Comment:
Could this perhaps be simplified as `output.len().min(c_uint::MAX as usize) as c_uint`?

Also, does the handling of `input.len()` also need to be updated above as well?

Reply:
`Could this perhaps be simplified as output.len().min(c_uint::MAX as usize) as c_uint?`
Yes.

`Also, does the handling of input.len() also need to be updated above as well?`
Yes.

I'll issue a patch accordingly in a moment.